src/aleph/http/core.clj

```diff
@@ -633,7 +633,7 @@
 (defn attach-idle-handlers [^ChannelPipeline pipeline idle-timeout]
   (if (pos? idle-timeout)
     (doto pipeline
-      (.addLast "idle" ^ChannelHandler (IdleStateHandler. 0 0 idle-timeout TimeUnit/MILLISECONDS))
+      (.addFirst "idle" ^ChannelHandler (IdleStateHandler. 0 0 idle-timeout TimeUnit/MILLISECONDS))
       (.addLast "idle-close" ^ChannelHandler (close-on-idle-handler)))
```

Comment:
Does idle-close need to be moved too?

Reply:
No, it doesn't have to.
The important part is the `IdleStateHandler` to be at the beginning of the `ChannelPipeline` (the closest to the `Socket.read` and the writes to the `ChannelHandlerContext`. [1]

`idle-close` only implements what to do when an `IdleState` is detected and can be anywhere on the pipeline.

[1] : https://netty.io/4.1/api/io/netty/channel/ChannelPipeline.html

Reply:
Can we safely group them together? It feels tidier that way.

Reply:
I finally moved the `attach-idle-handlers`